src/client/components/Signup.tsx

```diff
@@ -26,16 +44,25 @@ export const SignUp: React.FC<SignUpProps> = () => {
 
     return(
         <div>
-            <h3>SignUp</h3>
-            
-            <Card>
-                <CardContent>
-                    <TextField label="User Name" onChange={onTextFieldUpdate(setUserName)}></TextField>
-                    <Button onClick={()=>{printData(name, userName, password, email)}}>
-                        Test
-                    </Button>
-                </CardContent>
-            </Card>
+            <Paper square>
+                <Card>
+                    <CardContent>
+                        <Tabs indicatorColor="primary" textColor="primary" value={1}>
+                            <Link href='/login'><Tab label="Login" /></Link>
+                            <Tab label="Sign Up" />
+                        </Tabs>
+                        <TextField label="Name" onChange={onTextFieldUpdate(setName)}></TextField>
+                        <TextField label="Username" onChange={onTextFieldUpdate(setUserName)}></TextField>
+                        <TextField error={!properEmail()} label="Email" onChange={onTextFieldUpdate(setEmail)} helperText={!properEmail() ?  'Email is Invalid' : ''}></TextField>
+                        <TextField error = {!areSame(confirmedEmail, email)} label="Confirm Email" onChange={onTextFieldUpdate(setConfirmedEmail)} helperText={!areSame(confirmedEmail, email) ? 'Emails do not match' : ''}></TextField>
+                        <TextField error = {!properPassword()} label="Password" onChange={onTextFieldUpdate(setPassword)} helperText={!properPassword() ? 'Password must be between 9 and 255 characters' : ''}></TextField>
+                        <TextField error = {!areSame(confirmedPassword, password)} label="Confirm Password" onChange={onTextFieldUpdate(setConfirmedPassword)} helperText={!areSame(confirmedPassword, password) ? 'Passwords do not match' : ''}></TextField>
```

Comment:
Add {type: "password"} to the InputProps to make the passwords hidden when typing them
https://material-ui.com/api/text-field/
https://material-ui.com/api/input/

Reply:
ah forgot about this one!